indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/PendingTaskBasedWorkerProvisioningStrategy.java

```diff
@@ -279,30 +330,31 @@ private int getWorkersNeededToAssignTasks(
     public synchronized boolean doTerminate()
     {
       Collection<ImmutableWorkerInfo> zkWorkers = runner.getWorkers();
-      final WorkerBehaviorConfig workerConfig = workerConfigRef.get();
+      log.debug("Workers: %d [%s]", zkWorkers.size(), zkWorkers);
+      final WorkerBehaviorConfig workerConfig = getWorkerBehaviorConfig(workerConfigRef, "terminate", log);
       if (workerConfig == null) {
-        log.warn("No workerConfig available, cannot terminate workers.");
```

Comment:
Does this need to go away?

Reply:
This is built-in `getWorkerBehaviorConfig()` now